shared/src/vyxal/Exceptions.scala

```diff
@@ -113,6 +113,35 @@ class UnimplementedOverloadException(element: String, args: Seq[VAny])
 class VyxalRecursionException()
     extends VyxalRuntimeException("Too many recursions")
 
+class IterificationOfNonIterableException(value: VAny)
+    extends VyxalRuntimeException(s"Cannot iterify $value")
+
+class BadArgumentException(message: String, arg: VAny)
+    extends VyxalRuntimeException(s"$message received bad argument: $arg")
+
+/** Class related exceptions */
```

Comment:
We could make a `class VyxalClassException(msg: String) extends VyxalRuntimeException(msg)`. Not sure if we'll ever need to catch every kind of class-related exception though

Reply:
I think it's fine the way it is - only the specific exceptions I defined are used.